photon-core/src/main/java/org/photonvision/vision/camera/CameraInfo.java

```diff
@@ -80,15 +81,27 @@ public Optional<String> getUSBPath() {
     }
 
     @Override
-    public boolean equals(Object o) {
-        if (o == this) return true;
-        if (!(o instanceof UsbCameraInfo || o instanceof CameraInfo)) return false;
-        UsbCameraInfo other = (UsbCameraInfo) o;
-        return path.equals(other.path)
-                // && a.dev == b.dev (dev is not constant in Windows)
-                && name.equals(other.name)
-                && productId == other.productId
-                && vendorId == other.vendorId;
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj == null) return false;
+        if (getClass() != obj.getClass()) return false;
+        CameraInfo other = (CameraInfo) obj;
+
+        // Windows device number is not significant. See
+        // https://github.com/wpilibsuite/allwpilib/blob/4b94a64b06057c723d6fcafeb1a45f55a70d179a/cscore/src/main/native/windows/UsbCameraImpl.cpp#L1128
+        if (!Platform.isWindows()) {
+            if (dev != other.dev) return false;
+        }
+
+        if (!path.equals(other.path)) return false;
+        if (!name.equals(other.name)) return false;
+        if (!Arrays.equals(this.otherPaths, other.otherPaths)) return false;
```

Comment:
Is the order of the to arrays guaranteed to be the same?

Reply:
No, but I have changed to not need that